docs/registry/index.rst

```diff
@@ -4,48 +4,299 @@
 Registry (`pyvo.registry`)
 **************************
 
-This subpackage let you find data access services using search parameters.
+This is an interface to the Virtual Observatory Registry, a collection
+of metadata records of the VO's “resources” (“resource” is jargon for: a
+collection of datasets, usually with a service in front of it).  For a
+wider background, see `2014A&C.....7..101D`_ for the general
+architecture and `2015A&C....10...88D`_ for the search interfaces.
 
-Getting started
+.. _2014A&C.....7..101D: https://ui.adsabs.harvard.edu/abs/2014A%26C.....7..101D/abstract
+.. _2015A&C....10...88D: https://ui.adsabs.harvard.edu/abs/2015A%26C....10...88D/abstract
+
+There are two fundamental modes of searching in the VO:
+
+(a) Data discovery: This is when you are looking for some sort of data
+    collection based on its metadata; a classical example would be
+    something like “I need redshifts of supernovae”.
+
+(b) Service discovery: This is what you need when you want to query all 
+    services of a certain kind (e.g., „all spectral services claiming to
+    have infrared data“), which in turn is the basis of all-VO *dataset*
+    discovery (“give me all infrared spectra of 3C273”)
+
+Both modes are supported by this module.
+
+
+Basic interface
 ===============
-Registry searches are performed using the :py:meth:`pyvo.registry.search`
-method.
 
->>> from pyvo.registry import search as regsearch
+The main interface for the module is :py:meth:`pyvo.registry.search`;
+the examples below assume::
+
+  >>> from pyvo import registry
+
+This function accepts one or more search constraints, which can be
+either specificed using constraint objects as positional arguments or as
+keyword arguments.  The following constraints are available:
+
+* :py:class:`pyvo.registry.Freetext` (``keywords``): one or more
```

Comment:
All public API has an auto-generated summary from their docstrings, see at the bottom of the generated docs: https://pyvo--289.org.readthedocs.build/en/289/registry/index.html#reference-api

So I think this list is not really necessary here, and it may be better to just link to that section (and has the small risk of going out of date if/when new classes are being added).